chromadb/test/utils/test_embedding_function_schemas.py

```diff
@@ -0,0 +1,436 @@
+            "api_key_env_var": "CHROMA_JINA_API_KEY",
+        },
+        "config": {
+            "api_key_env_var": "CHROMA_JINA_API_KEY",
+            "model_name": "jina-embeddings-v2-base-en",
+        },
+    },
+    "voyageai": {
+        "args": {
+            "api_key": "dummy_key",
+            "model_name": "voyage-2",
+            "api_key_env_var": "CHROMA_VOYAGEAI_API_KEY",
+        },
+        "config": {
+            "api_key_env_var": "CHROMA_VOYAGEAI_API_KEY",
+            "model_name": "voyage-2",
+        },
+    },
+    "onnx_mini_lm_l6_v2": {
+        "args": {
+            "preferred_providers": ["onnxruntime"],
+        },
+        "config": {
+            "preferred_providers": ["onnxruntime"],
+        },
+        "mocks": [
+            ("onnxruntime", MagicMock()),
+        ],
+    },
+    "open_clip": {
+        "args": {
+            "model_name": "ViT-B-32",
+            "checkpoint": "laion2b_s34b_b79k",
+            "device": "cpu",
+        },
+        "config": {
+            "model_name": "ViT-B-32",
+            "checkpoint": "laion2b_s34b_b79k",
+            "device": "cpu",
+        },
+    },
+    "roboflow": {
+        "args": {
+            "api_key": "dummy_key",
+            "api_key_env_var": "CHROMA_ROBOFLOW_API_KEY",
+        },
+        "config": {
+            "api_key_env_var": "CHROMA_ROBOFLOW_API_KEY",
+        },
+    },
+    "text2vec": {
+        "args": {
+            "model_name": "shibing624/text2vec-base-chinese",
+        },
+        "config": {
+            "model_name": "shibing624/text2vec-base-chinese",
+        },
+    },
+}
+
+# Skip these embedding functions in tests as they require complex setup
+SKIP_EMBEDDING_FUNCTIONS = [
+    "huggingface_server",  # Requires a running server
+    "chroma_langchain",  # Requires LangChain setup
+    "default",  # Special case that delegates to ONNXMiniLM_L6_V2
+    "google_vertex",  # Requires complex mocking of vertexai module
+    "amazon_bedrock",  # Requires complex mocking of boto3 session
+]
+
+
+def test_all_schemas_are_valid_json() -> None:
+    """Test that all schemas are valid JSON"""
+    schema_names = get_all_schema_names()
+    for schema_name in schema_names:
+        # This will raise an exception if the schema is not valid JSON
+        schema: Dict[str, Any] = load_schema(schema_name)
+        assert isinstance(schema, dict)
+        assert "$schema" in schema
+        assert "title" in schema
+        assert "description" in schema
+        assert "version" in schema
+        assert "properties" in schema
+
+
+# Get embedding function names for parametrization
+def get_embedding_function_names() -> List[str]:
+    """Get all embedding function names to test"""
+    return [
+        name
+        for name in known_embedding_functions.keys()
+        if name not in SKIP_EMBEDDING_FUNCTIONS
+    ]
+
+
+@pytest.mark.usefixtures("embedding_function_dependencies")
+class TestEmbeddingFunctions:
```

Comment:
nit: curious if there's a reason for making this a class? the vast majority of our current tests are just functions